File: 백준/Silver/2567. 색종이 － 2/색종이 － 2.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int tc = Integer.parseInt(br.readLine());
        
        int[][] board = new int[102][102];
        for(int t = 0;t<tc;t++){
            StringTokenizer st = new StringTokenizer(br.readLine());
            int x = Integer.parseInt(st.nextToken());
            int y = Integer.parseInt(st.nextToken());

            for(int i = x;i<x+10;i++){
                for(int j = y;j<y+10;j++){
                    board[i][j] = 1;
                }
            }
        }
        int [][]directions = { {0, 1}, {0, -1}, {1, 0}, {-1, 0} };
        int res = 0;
        for(int i = 0;i<102;i++){
            for(int j = 0;j<102;j++){
                if(board[i][j]==1){
                    for(int[]d : directions){
                        int ni = i+d[0];
                        int nj = j+d[1];

                        if(ni<0 ||ni>=102||nj<0 ||nj>=102|| board[ni][nj]==0){
                            res+=1;
                        }
                    }
                }
            }
        }
        System.out.println(res);
    }
    
}

File: 백준/Silver/2567. 색종이 － 2/색종이 － 2.py
import sys
input = sys.stdin.readline

tc = int(input())
board = [[0]*102 for _ in range(102)] #도화지 크기를 여유롭게

#색종이 영역 표시
for _ in range(tc):
    x,y = map(int,input().split())
    for i in range(x,x+10):
        for j in range(y,y+10):
            board[i][j] = 1

directions = [(0,1),(0,-1),(1,0),(-1,0)]
res = 0
for i in range(102):
    for j in range(102):
        if board[i][j]==1: #색종이부분만 탐색
            for dx, dy in directions:
                ni,nj = i+dx, j+dy
                if board[ni][nj]==0:
                    res +=1
print(res)
